src/vcpkg/base/messages.cpp

```diff
@@ -0,0 +1,288 @@
+#include <vcpkg/base/json.h>
+#include <vcpkg/base/messages.h>
+#include <vcpkg/base/system.debug.h>
+
+namespace vcpkg::msg
+{
+    DECLARE_AND_REGISTER_MESSAGE(NoLocalizationForMessages, (), "", "No localization for the following messages:");
+
+    // basic implementation - the write_unlocalized_text_to_stdout
+#if defined(_WIN32)
+    static bool is_console(HANDLE h)
+    {
+        DWORD mode = 0;
+        // GetConsoleMode succeeds iff `h` is a console
+        // we do not actually care about the mode of the console
+        return GetConsoleMode(h, &mode);
+    }
+
+    static void check_write(BOOL success)
+    {
+        if (!success)
+        {
+            ::fwprintf(stderr, L"[DEBUG] Failed to write to stdout: %lu\n", GetLastError());
+            std::abort();
+        }
+    }
+    static DWORD size_to_write(::size_t size)
+    {
+        return size > static_cast<DWORD>(-1) // DWORD_MAX
+                   ? static_cast<DWORD>(-1)
+                   : static_cast<DWORD>(size);
```

Comment:
```suggestion
        return size > DWORD_MAX
                   ? DWORD_MAX
                   : static_cast<DWORD>(size);
```

You already say `DWORD` so `DWORD_MAX` should be available here. Alternately, replace the `DWORD`s with `unsigned long`?